### ISSUE TITLE
Error on macOS: `Cannot write headers after they are sent to the client`

### ISSUE THREAD

**Issue by josephmturner** [RangerMauve/hyper-gateway#20]
Repost of [this bug report](https://yhetil.org/emacs-devel/m1a5slqyqz.fsf@eshelyaron.com/):

```
FYI when trying this on macOS I had some issues with the `hyper-gateway`
program, it basically fails and quits after most hyperdrive.el commands.
For example, when I call `hyperdrive-hyper-gateway-version` it prints:

--8<---------------cut here---------------start------------->8---
Request: {
  method: 'GET',
  url: '/',
  headers: {
    host: 'localhost:4973',
    'user-agent': 'curl/8.1.2',
    accept: '*/*',
    'accept-encoding': 'deflate, gzip'
  }
}
Requested: {
  method: 'GET',
  url: 'hyper://',
  status: 500,
  responseHeaders: { 'content-type': 'text/plain; charset=utf-8' }
}
Error [ERR_HTTP_HEADERS_SENT]: Cannot write headers after they are sent to the client
    at new NodeError (node:internal/errors:406:5)
    at ServerResponse.writeHead (node:_http_server:345:11)
    at Server.<anonymous> (file:///usr/local/lib/node_modules/hyper-gateway/src/index.js:131:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
node:internal/errors:497
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot write headers after they are sent to the client
    at new NodeError (node:internal/errors:406:5)
    at ServerResponse.writeHead (node:_http_server:345:11)
    at Server.<anonymous> (file:///usr/local/lib/node_modules/hyper-gateway/src/index.js:139:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
--8<---------------cut here---------------end--------------->8---
```

**Comment by RangerMauve** [comment]
What version of the gateway of this was it?

**Comment by josephmturner** [comment]
> What version of the gateway of this was it?

Likely 3.4.2

**Comment by josephmturner** [comment]
This particular error fixed in [3.6.0](https://github.com/RangerMauve/hyper-gateway/releases/tag/v3.6.0).
